pkg/sql/type_change.go

```diff
@@ -791,6 +795,49 @@ func (t *typeSchemaChanger) canRemoveEnumValue(
 		query.WriteString(fmt.Sprintf("SELECT %s FROM [%d as t] WHERE", columns, ID))
 		firstClause := true
 		validationQueryConstructed := false
+
+		for _, idx := range desc.AllIndexes() {
```

Comment:
I'm curious as to why we can't restrict ourselves to `desc.NonDropIndexes()`. I suppose there's a good reason but I don't know it.

Reply:
That seems legit. I should add a test that I can drop the index and then the enum value in the same transaction.

Reply:
Nah, I don't want that because if we fail to drop the index, because we fail on the schema change, I don't want to corrupt the table. The darn concurrency and rollback bugs are hard to think about.